cmd/clientapi/main.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"flag"
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/notary/client_api/api"
+	"github.com/docker/notary/cmd/clientapi/setup"
+	"github.com/docker/notary/utils"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+var (
+	configPath string
+)
+
+func init() {
+	flag.StringVar(
+		&configPath,
+		"config",
+		"config.toml",
+		"path to configuration file; supported formats are JSON, YAML, and TOML",
+	)
+	flag.Parse()
+}
+
+func main() {
+	vc, err := setup.ViperConfig(configPath)
+	if err != nil {
+		logrus.Fatal(err)
+	}
+
+	logrus.SetLevel(logrus.DebugLevel)
+
+	// FIXME: refactor with notary-signer/config.go:getAddrAndTLSConfig()
+	// and setupGRPCServer()
+	grpcAddr, tlsConfig, err := setup.GetAddrAndTLSConfig(vc)
+	if err != nil {
+		logrus.Fatal("unable to set up TLS: %s", err.Error())
+	}
+
+	upstreamAddr := vc.GetString("upstream.addr")
+	upstreamCAPath := utils.GetPathRelativeToConfig(vc, "upstream.tls_ca_file")
+
+	creds := credentials.NewTLS(tlsConfig)
+	auth, err := setup.Authorization(vc)
```

Comment:
Another auth note - for the case when the client server (is there not a better name?) is NOT running locally, it would really be great to be able to provide auth as a plugin. For Quay we use the same token format but verify it differently (the server keeps an updated list of valid signing keys instead of using a CA)...Serverside I had to wrap notary server to support running with Quay, would be great to just swap out the plugin.

Obviously Quay is just an example, for notary to be useful to non-container-registries it will need to be able to support different auth mechanisms.